### DISCUSSION TITLE
SSL using docker  Nginx Proxy not work correctly

### DISCUSSION THREAD

**Discussion by drkrap** [firefly-iii/firefly-iii#6656] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I tried all the config show on the documentation about Docker and proxy connections using nginx but after I upgrade I cannot  run fireflyIII using SSL cause I get this error message from the browser: Refused to set the document's base URI to 'http://finanzas.casa.lan/' because it violates the following Content Security Policy directive: "base-uri 'self'".

Im using  Docker and this is my docker compose: 

```
version: '3.3'
services:
  app:
    image: fireflyiii/core:latest
    restart: always
    volumes:
      - firefly_iii_upload:/var/www/html/storage/upload
    env_file: .env
    ports:
      - 8085:8080
    depends_on:
      - db
  db:
    image: mariadb    
    hostname: fireflyiiidb
    restart: always
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_USER=firefly
      - MYSQL_PASSWORD=secret_firefly_password
      - MYSQL_DATABASE=firefly
    volumes:
      - firefly_iii_db:/var/lib/mysql
volumes:
   firefly_iii_upload:
   firefly_iii_db:

```

My ENV file

```
...
APP_URL=https://finanzas.casa.lan
TRUSTED_PROXIES=**

```
also I'm atthaing one config from the nginx proxy manager 
![Captura de Pantalla 2022-11-18 a la(s) 09 57 11](https://user-images.githubusercontent.com/6935022/202733985-2d721101-9919-4979-8271-0bb87eabb910.png)
![Captura de Pantalla 2022-11-18 a la(s) 09 56 00](https://user-images.githubusercontent.com/6935022/202734059-0be7b378-8b1b-4c8a-b11a-e7790f896122.png)
![Captura de Pantalla 2022-11-18 a la(s) 09 56 16](https://user-images.githubusercontent.com/6935022/202734060-42b26a1d-e87f-42be-811b-0623944eaf3a.png)

this issue start after I upgrade the containers from version : 5.7.14 to 5.7.15

### Expected behaviour

use proxy for SSL on docker instances + nginx proxy manager  protect the deployment using SSL HTTPS instead of http.

### Debug information

ebug information generated at 2022-11-18 10:44:23 America/Lima for Firefly III version **5.7.15**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.15 |
| Firefly III API | 1.5.6 |
| Build | 298, 02-11-2022 05:00:04 CET |
| DB version | 18 (exp. 18) |
| PHP | `8.1.11` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | America/Lima |
| Browser TZ | America/Lima |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-11-01 00:00:00 |
| Session end | 2022-11-30 23:59:59 |
| Session first | 2017-03-16 00:00:00 |
| User ID | 1 |
| User language | es_ES |
| User locale | es_ES |
| Attempt at "es_ES.utf8" | 'es_ES.utf8' |
| Attempt at "es_ES.UTF-8" | 'es_ES.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 Edg/107.0.1418.42 |

```
Truncated from this point <----|/html/app/Http/Controllers/Admin/UpdateController.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(162): FireflyIII\Http\Controllers\Admin\UpdateController::FireflyIII\Http\Controllers\Admin\{closure}()
#27 /var/www/html/app/Http/Controllers/Controller.php(112): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(162): FireflyIII\Http\Controllers\Controller->FireflyIII\Http\Controllers\{closure}()
#29 /var/www/html/app/Http/Middleware/Binder.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\Binder->handle()
#31 /var/www/html/app/Http/Middleware/Range.php(62): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\Range->handle()
#33 /var/www/html/app/Http/Middleware/IsAdmin.php(62): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\IsAdmin->handle()
#35 /var/www/html/app/Http/Middleware/Authenticate.php(75): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\Authenticate->handle()
#37 /var/www/html/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Laravel\Passport\Http\Middleware\CreateFreshApiToken->handle()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Session\Middleware\AuthenticateSession->handle()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#42 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
#43 /var/www/html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#44 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\View\Middleware\ShareErrorsFromSession->handle()
#45 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#46 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest()
#47 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Session\Middleware\StartSession->handle()
#48 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#49 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
#50 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#51 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\EncryptCookies->handle()
#52 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#53 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(726): Illuminate\Pipeline\Pipeline->then()
#54 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(703): Illuminate\Routing\Router->runRouteWithinStack()
#55 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(667): Illuminate\Routing\Router->runRoute()
#56 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(656): Illuminate\Routing\Router->dispatchToRoute()
#57 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(190): Illuminate\Routing\Router->dispatch()
#58 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
#59 /var/www/html/app/Http/Middleware/InstallationId.php(52): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#60 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\InstallationId->handle()
#61 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#62 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Http\Middleware\TrustProxies->handle()
#63 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#64 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#65 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle()
#66 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#67 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#68 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\TrimStrings->handle()
#69 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#70 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
#71 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#72 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
#73 /var/www/html/app/Http/Middleware/SecureHeaders.php(51): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#74 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\Http\Middleware\SecureHeaders->handle()
#75 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#76 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\Pipeline\Pipeline->then()
#77 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(134): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
#78 /var/www/html/public/index.php(75): Illuminate\Foundation\Http\Kernel->handle()
#79 {main}  
[2022-11-18 10:17:26] local.ERROR: Update information contains an error.  
[2022-11-18 10:17:26] local.ERROR: Guzzle: cURL error 28: Resolving timed out after 3141 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://version.firefly-iii.org/index.json  

```

### Steps to reproduce

1.  install a clean firefly deployment using steps on : https://docs.firefly-iii.org/firefly-iii/installation/docker/
2.  install a clean nginx-proxy-manager  using steps on : https://nginxproxymanager.com/setup/ 
3.  create a certificate and set a proxy for firefly on nginx
4. try to login on nginx, press f12 on your browser, check the console, it will show a similar message : Refused to set the document's base URI to 'http://finanzas.casa.lan/' because it violates the following Content Security Policy directive: "base-uri 'self'".

**Comment by JC5** [comment]
This happens when the proxy manager doesn't forward the necessary headers.

```
proxy_set_header X-Forwarded-Host $host;
proxy_set_header X-Forwarded-Server $host;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header Host $host;
client_max_body_size 64M;
proxy_read_timeout 300s;
```

**Reply by drkrap** [reply]
I put all the command on the nginx proxy manager And I get this error
![Captura de Pantalla 2022-11-18 a la(s) 13 37 42](https://user-images.githubusercontent.com/6935022/202778600-994d160e-5dd1-4a05-b3fb-93dd76fad360.png)

deleting this, `proxy_set_header Host $host;` resolve but I still have the issue:
![Captura de Pantalla 2022-11-18 a la(s) 13 39 46](https://user-images.githubusercontent.com/6935022/202779028-5c3b6e04-63d7-4bef-823d-146be9916e0e.png)

here is my nginx config:

![Captura de Pantalla 2022-11-18 a la(s) 13 42 26](https://user-images.githubusercontent.com/6935022/202779356-8c6e0d0a-f0e4-41bf-8604-b6bc3533e1d3.png)

**Comment by xzaz** [comment, marked as answer]
I'v fixed this by setting the ENV `TRUSTED_PROXIES` to `**`

**Reply by brimell** [reply]
Wow ok, thanks so much. That actually took forever for me to find this thread

**Comment by V373z** [comment]
I have the same problem. Docker is hosted on Unraid. The app runs on port 8080 in NPM. I assigned the IP address and port and my domain, but the app is not accessible via the domain. I also set the advanced settings

that's all I've googled so far.
location / {
proxy_set_header X-Forwarded-Host $host:$port;
proxy_set_header X-Forwarded-Server $host:$port;
proxy_set_header X-Forwarded-Host $host;
proxy_set_header X-Forwarded-Server $host;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header Host $host;
client_max_body_size 64M;
proxy_read_timeout 300s;
}

TRUSTED_PROXIES: **

I had the app running about 3-4 years ago and was also accessible via NPM.
What else could/should I check?

**Reply by V373z** [reply]
My problem was that port 80 was not open on NPM.
